### PR TITLE
README: add MacPorts instructions

### PR DESCRIPTION
**Pull request by herbygillot**

### CODE REVIEW

**Comment by TheLostLambda** [comment]
I'll pester @qballer who runs macOS just to be sure this is working, but this looks cool!

**Comment by qballer** [comment]
I'm not a macports user, to be honest. how about @5c077m4n  ?

**Comment by 5c077m4n** [comment]
Sorry, I use homebrew too... @qballer

**Comment by SaintFenix** [comment]
I've got a Mac Pro, I'll see if I can spin up a VM to test it out in real quick.

**Comment by SaintFenix** [comment]
Alrighty, I got Macports installed, and setup, boy is it not the most straightforward thing compared to homebrew, especially on an older MacOS. I used MacOS High Sierra, and was able to install it with no issue with Macports, everything functioned as normal, and I wasn't able to get any issues in the process.

Though notably, on a fresh install of Macports, installing Zellij will have you download quite a few dependencies. I'm not versed enough in Macports to know if that is just the typical Rust toolchain being built to compile it, or if its excessive.

But all in all it works. 👍

**Comment by herbygillot** [comment]
> Alrighty, I got Macports installed, and setup, boy is it not the most straightforward thing compared to homebrew, especially on an older MacOS. I used MacOS High Sierra, and was able to install it with no issue with Macports, everything functioned as normal, and I wasn't able to get any issues in the process.
> 
> Though notably, on a fresh install of Macports, installing Zellij will have you download quite a few dependencies. I'm not versed enough in Macports to know if that is just the typical Rust toolchain being built to compile it, or if its excessive.
> 
> But all in all it works. 👍

Thank you for testing, @SaintFenix.  I'm curious though; what were some of things that you felt weren't straightforward while setting up MacPorts?

And dependencies will be downloaded only if a binary package of `zelij` wasn't in MacPorts' repo yet.  That's done by MacPorts build infra.  Once that's in-place, then just the `zellij` package gets downloaded and installed.  Wondering why that wasn't the case for you.

**Comment by SaintFenix** [comment]
>  I'm curious though; what were some of things that you felt weren't straightforward while setting up MacPorts?

It's outside of this pull request, but,

Homebrew is a one liner install, whereas Macports needs Xcode, which you have to download from the Mac App store. Not a huge issue, but if you are on the older version of the OS, you can't do it from the Mac App Store, you have to get (or already have) a developer account at apple to download and older version for your OS. So that's a bit of googling and setting up, and downloading. Once setup, you accept the licenses, and find the right package, install Macports, and then update the ports, and then you are ready to install.

This took me around 10-15 minutes, whereas homebrews install is 2-3.

It's not really a showstopper, but I'm just used to Homebrew's one line.

That aside, I'm still quite happy that Zellij is being included in Macports. I appreciate your Pull Request on it.

**Comment by herbygillot** [comment]
> It's outside of this pull request, but,
> 
> Homebrew is a one liner install, whereas Macports needs Xcode, which you have to download from the Mac App store. Not a huge issue, but if you are on the older version of the OS, you can't do it from the Mac App Store, you have to get (or already have) a developer account at apple to download an older version for your OS with the Command line pack seperate. So that's a bit of googling and setting up, and downloading. Once setup, you accept the licenses, and find the right package, install Macports, and then update the ports, and then you are ready to install.
> 
> This took me around 10-15 minutes, whereas homebrews install is 2-3.
> 
> It's not really a showstopper, but I'm just used to Homebrew's one line.
> 
> That aside, I'm still quite happy that Zellij is being included in Macports. I appreciate your Pull Request on it.

Thank you and much appreciated, @SaintFenix

**Comment by 5c077m4n** [comment]
@SaintFenix According to https://docs.brew.sh/Installation#macos-requirements it says that homebrew needs Xcode too

**Comment by SaintFenix** [comment]
> @SaintFenix According to https://docs.brew.sh/Installation#macos-requirements homebrew needs Xcode too

Ah so that is the case, Macports gets a pass.

**Comment by qballer** [comment]
@herbygillot thanks a lot for creating this. This seems very cool.  I'm sorry I'm taking my time with merging this, but IMO the [portfile ](https://github.com/macports/macports-ports/blob/5842e76e83e75d057b0e309a042b9befe9e1526f/sysutils/zellij/Portfile) currently poses some kind of a security threat as you aren't using the `cargo.toml` in this repo to fetch crates nor the binaries published.

Now, I know very little about macports, if there is something I'm missing do let me know because the last thing I want to do is gatekeeping.

**Comment by herbygillot** [comment]
No worries @qballer, that's a 100% valid concern, you're absolutely in the right to call that out.

So to start, all software in MacPorts are ports that are defined by their Portfiles.

A Portfile defines:
1. project info (name, description, version, homepage, license, supported architectures, etc.)
2. the names of the downloadable source files (distname)
3. how to verify the source files (checksums)
4. how to build it
5. how to install it into the MacPorts prefix (/opt/local)

Note 1: Many of these options are implicit and/or default, so they aren't always going to be defined in a Portfile.

Note 2: any downloadable source files defined are automatically uploaded to MacPorts mirrors so that the software can still be built even if the original upstream repo disappears.

A Portfile by default wants to build things in the typical _./configure_, _make_, _make install_ way, but through the use of Portgoups (or by explicitly providing build commands), we support different packaging methods and programming languages.  For `zellij`, the `cargo` Portgroup will have MacPorts build `zellij` using Rust & cargo.  The `cargo` Portgroup exposes `cargo.crates`.

`cargo.crates` is like a macro that defines Crate dependencies as MacPorts distfiles so that MacPorts can download and mirror the crates.  In this way as mentioned above, MacPorts can still build `zellij` even if crates cannot be downloaded from upstream.

The `cargo.crates` section is defined directly from `zellij`'s Cargo.toml, and will pull crates directly from crates.io.

You can see this here:
https://github.com/macports/macports-ports/blob/cf021e88a60d8c4bd2b6a852796b1d224e40dec8/_resources/port1.0/group/cargo_fetch-1.0.tcl#L79

Additionally, the SHA256 checksum you see there is the checksum of the crate exactly as it is downloaded from https://crates.io

You can test this yourself:

If you put together the URL pattern for `zellij`'s first dep, [`addr2line`](https://github.com/macports/macports-ports/blob/cf021e88a60d8c4bd2b6a852796b1d224e40dec8/sysutils/zellij/Portfile#L62), you'll get the following URL:

https://crates.io/api/v1/crates/addr2line/0.14.1/download?dummy=:crate-addr2line-a55f82cfe485775d02112886f4169bde0c5894d75e79ead7eafe7e40a25e45f7

After downloading the crate from that URL, its checksum should match what you see in `zellij`'s Portfile:

```
❯ wget -O addr2line-crate 'https://crates.io/api/v1/crates/addr2line/0.14.1/download?dummy=:crate-addr2line-a55f82cfe485775d02112886f4169bde0c5894d75e79ead7eafe7e40a25e45f7'
--2021-04-26 15:10:51--  https://crates.io/api/v1/crates/addr2line/0.14.1/download?dummy=:crate-addr2line-a55f82cfe485775d02112886f4169bde0c5894d75e79ead7eafe7e40a25e45f7
Resolving crates.io (crates.io)... 13.226.29.81, 13.226.29.57, 13.226.29.69, ...
Connecting to crates.io (crates.io)|13.226.29.81|:443... connected.
HTTP request sent, awaiting response... 302 Moved Temporarily
Location: https://static.crates.io/crates/addr2line/addr2line-0.14.1.crate [following]
--2021-04-26 15:10:52--  https://static.crates.io/crates/addr2line/addr2line-0.14.1.crate
Resolving static.crates.io (static.crates.io)... 54.192.100.97, 54.192.100.107, 54.192.100.42, ...
Connecting to static.crates.io (static.crates.io)|54.192.100.97|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 31201 (30K) [application/x-tar]
Saving to: ‘addr2line-crate’

addr2line-crate                              100%[============================================================================================>]  30.47K  --.-KB/s    in 0.009s

2021-04-26 15:10:52 (3.29 MB/s) - ‘addr2line-crate’ saved [31201/31201]

~ 
❯ shasum -a 256 addr2line-crate
a55f82cfe485775d02112886f4169bde0c5894d75e79ead7eafe7e40a25e45f7  addr2line-crate
~ 
❯ grep a55f82cfe485775d02112886f4169bde0c5894d75e79ead7eafe7e40a25e45f7 ~/Source/ports/sysutils/zellij/Portfile
    addr2line                       0.14.1  a55f82cfe485775d02112886f4169bde0c5894d75e79ead7eafe7e40a25e45f7 \
```

A Portfile has a maintainer, which is the typical model for most 3rd-party package repos (like the one for Archlinux or Ubuntu), and is responsible for updating the Portfile on software updates and and reviewing changes to proposed to the same.

**Comment by herbygillot** [comment]
Sorry, that was a lot but:

TLDR:
* `cargo.crates` in a Portfile is generated directly from a project's Cargo.lock
* `cargo.crates` exists so that crates can be specified in a MacPorts-visible way, and MacPorts can mirror them
* crates in `cargo.crates` are downloaded rom https://crates.io
* so for crates in the Portfile will verify assets coming from https://crates.io

**Comment by qballer** [comment]
After discussing this with some of the maintainers of Zellijs and going through all the information your provided we decided we are going to do a community maintained packages area in our website docs where people can add links/instruction to all sorts of packages (nix, arch, MacPorts... etc)

Adding this to the readme won't be possible at this time. If are willing to create another PR to our docs with this I will be more than happy to accept that.

**Comment by herbygillot** [comment]
Sure, that's fine. If you can link me to the docs in question, I'll get a PR up at some point.

**Comment by a-kenji** [comment]
@herbygillot

https://github.com/zellij-org/zellij-org.github.io
https://zellij.dev/documentation/introduction.html

The install section isn't there yet though.

**Comment by a-kenji** [comment]
@herbygillot,
Thank you for this pr.
We now have settled on a way to display third party repositories.
There is already a section about mac ports in there,
but maybe you want to add something to it?
https://github.com/zellij-org/zellij/pull/857
#857 
https://github.com/zellij-org/zellij/blob/main/THIRD_PARTY_INSTALL.md
